pkg/kubelet/kubelet.go

```diff
@@ -1778,6 +1790,12 @@ func (kl *Kubelet) syncTerminatingPod(ctx context.Context, pod *v1.Pod, podStatu
 		return err
 	}
 
+	// Once the containers are stopped, we can stop probing for liveness and readiness.
+	// TODO: once a pod is terminal, certain probes (liveness exec) could be stopped immediately after
+	//   the detection of a container shutdown or (for readiness) after the first failure. Tracked as
+	//   https://github.com/kubernetes/kubernetes/issues/107894 although may not be worth optimizing.
+	kl.probeManager.RemovePod(pod)
```

Comment:
Adding this invariant to the unit tests would be good.  SyncTerminatingPod should clear probe status, and if kill container returns an error, we should have only stopped liveness probes.

Reply:
Thank you for the feedback. This is a good ide and I added a second test to this PR.